lib/main.js

```diff
@@ -0,0 +1,244 @@
+  setInput("finalPct", "100")
+  setInput("monitors", "https://www.google.com")
+  setInput("errorThreshold", "0")
+}
+setInputs() // for local runs only to set environment variables
+// END TEST CODE
+
+// grab input values from github environment
+const APP = core.getInput('appName') // REQD application name
+const RG = core.getInput('resourceGroup') // REQD resource group name
+const IMAGE = core.getInput('imageID') // REQD image to update with
+const REV_SUFFIX = core.getInput('revisionSuffix') // REQD new revision suffix
+const LOG_ANALYTICS = core.getInput('logAnalyticsWorkspace') // log analytic workspace app is in
+const CANARY = core.getBooleanInput('canaryDeploy') // whether to use canary deployment
+const STEP_PCT = parseInt(core.getInput('stepPct')) // step pct of traffic going to new revision
+const STEP_TIME = parseFloat(core.getInput('stepTime')) // step time between traffic shifts
+const FINAL_PCT = parseInt(core.getInput('finalPct')) // final pct of traffic going to new revision
+const MONITORS = core.getInput('monitors').replace(/ +/g, '').split(/,|;|\n/) // api endpoints to hit
+const MONITOR_INTERVAL = parseFloat(core.getInput('monitorInterval')) // time between monitoring
+const ERROR_THRESHOLD = parseFloat(core.getInput('errorThreshold')) // max error tolerance on http pings
+const TIME_CONVERSION = 60000 // conversion factor from minutes to milliseconds
+
+let RESOURCE = ""
+let prevTrafficSettings = ""
+
+// utility function for async exec command
+const execute = function (command) {
+  return new Promise((resolve, reject) => {
+    console.log(`Running command ${command} at ${getDateTime()}`)
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        if (error.signal == 'SIGTERM') {
+          resolve('Process was killed');
+        } else {
+          reject(error);
+        }
+      } else {
+        core.debug(stdout)
+        core.debug(stderr)
+        resolve(stdout);
+      }
+    });
+  })
+}
+
+// sleep callback with delay in ms
+const sleep = (delay) => new Promise(r => setTimeout(r, delay));
+
+// get date and time for queries
+function getDateTime() {
+  var today = new Date();
+  var date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate();
+  var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+  return date + 'T' + time;
+}
+
+// monitor the new revision for the specified amount of time
+async function monitor() {
+  let guard = true
+  setTimeout(() => guard = false, STEP_TIME * TIME_CONVERSION)
+
+  let startTime = getDateTime()
+  while (guard) {
+    console.log(`Waiting for ${MONITOR_INTERVAL} ${MONITOR_INTERVAL == 1 ? "minute" : "minutes"}...`)
+    await sleep(MONITOR_INTERVAL * TIME_CONVERSION)
+    console.log(`Running tests...`)
+    await runTests(startTime)
+  }
+}
+
+// function to run tests
+async function runTests(startTime) {
+  // run user provided tests by pinging apis
+  if (MONITORS) {
```

Comment:
good to use more meaningful variable names for better readability of code.
I am guessing Monitors are api endpoints to hit (based on the comments at declaration)
If so, consider using the name 'apiEndpointsToTest'